firedrake/slate/slate.py

```diff
@@ -975,7 +976,7 @@ def subdomain_data(self):
                             "Domains must agree!"
                         )
 
-        return {self.ufl_domain(): sd}
+        return {extract_unique_domain(self): sd}
```

Comment:
I think all of the changes in this file need to be reverted, and then you need to track down the few cases where the call is being made against a UFL expression.

Reply:
I'll revert and will try my best to track it.